File: source_code/ranges_for_loop_filter_view_perf_example.cpp
static double transform_number(int input) {
    return input * 5;
}

int main() {
    static constexpr auto nums = std::array{1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12};
    operate_ranges(nums, 3, 45.0);
    return 0;
}

File: source_code/ranges_for_loop_filter_view_perf_example.rs
fn operate_ranges<T>(
    container: T, 
    offset: usize, 
    early_exit: f64, 
) where
    T: IntoIterator<Item = i32>,
{
    container.into_iter()
        .skip(offset)
        .filter(|&v| { is_odd(v) })
        .map(|v| { transform_number(v) })
        .take_while(|&v| { v != early_exit})
        .for_each(|v| print(v));
}